api/api.go

```diff
@@ -198,19 +202,23 @@ func (t *TLSConfig) Copy() *TLSConfig {
 	return nt
 }
 
-// DefaultConfig returns a default configuration for the client
-func DefaultConfig() *Config {
-	config := &Config{
-		Address:    "http://127.0.0.1:4646",
-		httpClient: cleanhttp.DefaultClient(),
```

Comment:
Previously, DefaultConfig initializes httpClient for client to use.  This PR changes it so default httpClient creation occurs in NewClient instead and have it be a field of the api client rather than config.